src/v/pandaproxy/schema_registry/requests/config.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "pandaproxy/json/rjson_parse.h"
+#include "pandaproxy/json/rjson_util.h"
+#include "pandaproxy/schema_registry/types.h"
+
+namespace pandaproxy::schema_registry {
+
+struct put_config_req_rep {
+    static constexpr std::string_view field_name = "compatibility";
+    compatibility_level compat{compatibility_level::none};
+};
+
+template<typename Encoding = rapidjson::UTF8<>>
+class put_config_handler : public json::base_handler<Encoding> {
+    enum class state {
+        empty = 0,
+        object,
+        compatibility,
+    };
+    state _state = state::empty;
+
+public:
+    using Ch = typename json::base_handler<Encoding>::Ch;
+    using rjson_parse_result = put_config_req_rep;
+    rjson_parse_result result;
+
+    explicit put_config_handler()
+      : json::base_handler<Encoding>{json::serialization_format::none}
+      , result() {}
+
+    bool Key(const Ch* str, rapidjson::SizeType len, bool) {
+        auto sv = std::string_view{str, len};
+        if (_state == state::object && sv == put_config_req_rep::field_name) {
```

Comment:
> I like this `put_config_req_rep::field_name` use here I think a few handlers hard coded the field name string in multiple places.

Most of them do, because there's not usually much chance of reuse; either due to the complexity or the coincidence that they have similar layout but are used for different purposes that are likely to evolve independently.